src/nft/components/collection/CollectionStats.tsx

```diff
@@ -264,21 +265,26 @@ const statsLoadingSkeleton = (isMobile: boolean) =>
   )
 
 const StatsRow = ({ stats, isMobile, ...props }: { stats: GenieCollection; isMobile?: boolean } & BoxProps) => {
-  const uniqueOwnersPercentage = stats.stats
-    ? roundWholePercentage((stats.stats.num_owners / stats.stats.total_supply) * 100)
-    : 0
-  const totalSupplyStr = stats.stats ? quantityFormatter(stats.stats.total_supply) : 0
+  const isNftGraphQl = useNftGraphQlFlag() === NftGraphQlVariant.Enabled
+  const uniqueOwnersPercentage =
+    stats.stats && stats.stats.total_supply
+      ? roundWholePercentage(((stats.stats.num_owners ?? 0) / stats.stats.total_supply) * 100)
+      : 0
+  const totalSupplyStr = stats.stats ? quantityFormatter(stats.stats.total_supply ?? 0) : 0
+  console.log(stats?.stats?.total_listings, stats?.stats?.total_supply)
   const listedPercentageStr =
-    stats.stats && stats.stats.total_listings > 0
-      ? roundWholePercentage((stats.stats.total_listings / stats.stats.total_supply) * 100)
+    stats.stats && stats.stats.total_supply
+      ? roundWholePercentage(((stats.stats.total_listings ?? 0) / stats.stats.total_supply) * 100)
       : 0
   const isCollectionStatsLoading = useIsCollectionLoading((state) => state.isCollectionStatsLoading)
 
   // round daily volume & floorPrice to 3 decimals or less
-  const totalVolumeStr = volumeFormatter(stats.stats?.total_volume)
-  const floorPriceStr = floorFormatter(stats.floorPrice)
+  const totalVolumeStr = volumeFormatter(stats.stats?.total_volume ?? 0)
+  const floorPriceStr = floorFormatter(stats.stats?.floor_price ?? 0)
   const floorChangeStr =
-    stats.stats && stats.stats.one_day_floor_change ? Math.round(Math.abs(stats.stats.one_day_floor_change) * 100) : 0
+    stats.stats && stats.stats.one_day_floor_change
+      ? Math.round(Math.abs(stats.stats.one_day_floor_change) * (isNftGraphQl ? 1 : 100))
```

Comment:
can you add a comment explaining the magic numbers here? would be good to have it tagged w/ a todo for removal if possible